### PR TITLE
Introducing revision date

### PR DESCRIPTION
**Pull request by xiam**
The `revisionDate` value will hold the date and time of the latest commit in `HEAD`.

The old `buildDate` value is still around because it *might* provide useful extra information, but in most cases it was just replaced by `revisionDate`

Closes https://github.com/getlantern/lantern/issues/2753

### CODE REVIEW

File: src/github.com/getlantern/flashlight/logging/logging.go [hunk 1]
```diff
@@ -133,7 +133,7 @@ func enableLoggly(addr string, cloudConfigCA string, instanceId string,
 	logglyWriter := &logglyErrorWriter{
 		lang:            lang,
 		tz:              time.Now().Format("MST"),
-		versionToLoggly: fmt.Sprintf("%v (%v)", version, buildDate),
+		versionToLoggly: fmt.Sprintf("%v (%v)", version, revisionDate),
```
Comment (xiam) [inline src/github.com/getlantern/flashlight/logging/logging.go]: This is the version string that is reported to loggly.

File: Makefile [hunk 2]
```diff
@@ -13,10 +13,14 @@ SVGEXPORT := $(shell which svgexport 2> /dev/null)
 
 BOOT2DOCKER := $(shell which boot2docker 2> /dev/null)
 
-BUILD_DATE := $(shell date -u +%Y%m%d.%H%M%S)
 GIT_REVISION := $(shell git describe --abbrev=0 --tags --exact-match 2> /dev/null || git rev-parse --short HEAD)
+GIT_REVISION_DATE := $(shell git show -s --format=%ci $(GIT_REVISION))
+
+REVISION_DATE := $(shell date -u -j -f "%F %T %z" "$(GIT_REVISION_DATE)" +"%Y%m%d.%H%M%S" || date -u -d "$(GIT_REVISION_DATE)" +"%Y%m%d.%H%M%S")
```
Comment (xiam) [inline Makefile]: The fallback is the Linux version of the command.

**Comment by xiam** [comment]
Hello @myleshorton,

Could you test this one?

Thanks.

**Comment by myleshorton** [comment]
Awesome thanks @xiam! I'll give this a look now.

**Comment by myleshorton** [comment]
LGTM @xiam thanks!

**Pull request merged**